src/pay/batch_payer.py

```diff
@@ -443,6 +484,25 @@ def attempt_single_batch(self, payment_records, op_counter, dry_run=None):
         if status != HTTPStatus.OK:
             logger.error("Error in forge operation")
             return PaymentStatus.FAIL, ""
+        size = SIGNATURE_BYTES_SIZE + len(bytes) / 2
+        required_fee = math.ceil(MINIMUM_FEE_MUTEZ + MUTEZ_PER_GAS_UNIT * total_gas + MUTEZ_PER_BYTE * size)
+        logger.info(f'minimal required fee is {required_fee}, current used fee is {total_fees}')
+
+        while total_fees < required_fee:
+            difference_fees = int(math.ceil(required_fee - total_fees))
+            first_tx = json.loads(content_list[0])
+            first_tx['fee'] = str(int(int(first_tx['fee']) + difference_fees))
+            total_fees = required_fee
+            content_list[0] = json.dumps(first_tx)
+            contents_string = ",".join(content_list)
+            forge_json = FORGE_JSON.replace('%BRANCH%', branch).replace("%CONTENT%", contents_string)
+            status, bytes = self.clnt_mngr.request_url_post(self.comm_forge, forge_json)
+            if status != HTTPStatus.OK:
+                logger.error("Error in forge operation")
+                return PaymentStatus.FAIL, ""
+            size = SIGNATURE_BYTES_SIZE + len(bytes) / 2
+            required_fee = math.ceil(MINIMUM_FEE_MUTEZ + MUTEZ_PER_GAS_UNIT * total_gas + MUTEZ_PER_BYTE * size)
```

Comment:
Is it possible to reuse this as a function with total gas and size as variables? Because it's already used above. It would be nice to have this block as a function to also test the output and have a comment what this block is doing.